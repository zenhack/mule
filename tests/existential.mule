#XFAIL: type formatting
(fn x. 4) : exist t. t -> t
