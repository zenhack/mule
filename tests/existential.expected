
Desugared:

	(: (fn x (Integer 4)) (exist t (t -> t)))

inferred type:

	(exist a (a -> a))

Runtime term:

	(Lam (0 () (Const (Integer 4))))

Evaluated:

	(Lam (0 () (Const (Integer 4))))
